test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/NestedJSClassTest.scala

```diff
@@ -0,0 +1,596 @@
+
+        def foo(a: String): String = xxx + yyy + abc + a
+      }
+
+      js.constructorOf[LocalJSClass]
+    }
+  }
+
+  class ScalaTraitContainerSubclass(val xxx: String) extends ScalaTraitContainer
+
+  class ScalaClassContainerWithObject(xxx: String) {
+    object InnerJSObject extends js.Object with TestInterface {
+      val zzz: String = xxx + "zzz"
+
+      def foo(a: String): String = xxx + "zzz" + a
+    }
+
+    def makeLocalJSObject(yyy: String): TestInterface = {
+      object LocalJSObject extends js.Object with TestInterface {
+        val zzz: String = xxx + yyy
+
+        def foo(a: String): String = xxx + yyy + a
+      }
+
+      LocalJSObject
+    }
+  }
+
+  class ScalaClassContainerWithSubclasses(val abc: String,
+      val parents: ScalaClassContainer) {
+
+    class InnerJSSubclass(yyy: String) extends parents.InnerJSClass(yyy) {
+      def foobar(): String = abc + yyy + zzz
+    }
+
+    def getInnerJSSubclass: js.Dynamic =
+      js.constructorOf[InnerJSSubclass]
+
+    def makeLocalJSSubclass(yyy: String): js.Dynamic = {
+      class LocalJSSubclass(xyz: String) extends parents.InnerJSClass(xyz) {
+        def foobar(): String = abc + yyy + zzz
+      }
+      js.constructorOf[LocalJSSubclass]
+    }
+  }
+
+  class ScalaClassContainerWithSubObjects(val abc: String,
+      val parents: ScalaClassContainer) {
+
+    object InnerJSObject extends parents.InnerJSClass(abc) {
+      def foobar(): String = abc + zzz
+    }
+
+    def makeLocalJSObject(yyy: String): js.Dynamic = {
+      object LocalJSObject extends parents.InnerJSClass(yyy) {
+        def foobar(): String = abc + yyy + zzz
+      }
+      LocalJSObject.asInstanceOf[js.Dynamic]
+    }
+  }
+
+  class GenericJSSuperClassContainer {
+    class GenericJSSuperClass[A, B <: List[Seq[A]]](val a: A, val b: B)
+        extends js.Object
+  }
+
+  class ScalaClassContainerWithTypeParameters[A](val a: A,
+      val parents: GenericJSSuperClassContainer) {
+
+    class GenericJSInnerClass[B <: List[Seq[A]]](b: B)
+        extends parents.GenericJSSuperClass[A, B](a, b)
+
+    def makeGenericJSLocalClass(): js.Dynamic = {
+      class GenericJSLocalClass[B <: List[Seq[A]]](b: B)
+          extends parents.GenericJSSuperClass[A, B](a, b)
+      js.constructorOf[GenericJSLocalClass[_]]
+    }
+
+    object GenericJSInnerObject
+        extends parents.GenericJSSuperClass[A, List[List[A]]](a, Nil)
+
+    def makeGenericJSInnerObject[B <: List[Seq[A]]](b: B): js.Dynamic = {
+      object GenericJSInnerObject
+          extends parents.GenericJSSuperClass[A, B](a, b)
+
+      GenericJSInnerObject.asInstanceOf[js.Dynamic]
+    }
+  }
+
+  class JSClassContainer(xxx: String) extends js.Object {
+    class InnerJSClass(yyy: String) extends js.Object with TestInterface {
+      val zzz: String = xxx + yyy
+
+      def foo(a: String): String = xxx + yyy + a
+    }
+
+    def getInnerJSClass: js.Dynamic =
+      js.constructorOf[InnerJSClass]
+  }
+
+}
```

Comment:
Because that's not allowed at compile-time. See
* https://github.com/sjrd/scala-js/blob/f82de5f73c49bec86bc1d4090403e559e1accf93/compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala#L778
* https://github.com/sjrd/scala-js/blob/f82de5f73c49bec86bc1d4090403e559e1accf93/compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala#L1268

That said I think I have somehow removed the test forbidding a native JS class/trait/object inside a non-native JS class/trait. In current master it's done in `JSInteropTest.notNested`, but that test was somehow restricted to `notNestedInScalaClassTrait`. There should also be `notNestedInNonNativeJSClassTrait`.